curve-amm/src/lib.rs

```diff
@@ -0,0 +1,1166 @@
+// CurveAmm Dex
+// Implementation based on https://github.com/equilibrium-eosdt/equilibrium-curve-amm/blob/master/pallets/equilibrium-curve-amm/src/lib.rs
+// For more details read
+//      https://miguelmota.com/blog/understanding-stableswap-curve/
+//      https://curve.fi/files/stableswap-paper.pdf
+//      https://github.com/equilibrium-eosdt/equilibrium-curve-amm/blob/master/docs/deducing-get_y-formulas.pdf
+
+#![no_std]
+#![feature(const_btree_new)]
+
+extern crate alloc;
+
+use alloc::collections::BTreeMap;
+use alloc::collections::BTreeSet;
+use alloc::vec;
+use alloc::vec::Vec;
```

Comment:
`use` section can be combined to be shorter. Just for example.

```suggestion
use alloc::collections::{BTreeMap, BTreeSet};
use alloc::vec::{self, Vec};
```

But we usually include all module related stuff with the one `use` entry.